rafx-assets/src/assets/graphics_pipeline/assets.rs

```diff
@@ -349,6 +350,51 @@ impl MaterialAsset {
     ) -> Option<usize> {
         self.inner.pass_phase_to_index.get(&index).copied()
     }
+
+    pub fn get_single_material_pass(&self) -> RafxResult<ResourceArc<MaterialPassResource>> {
+        if self.inner.passes.len() == 1 {
+            Ok(self
+                .inner
+                .passes
+                .get(0)
+                .unwrap()
+                .material_pass_resource
+                .clone())
+        } else {
+            Err(RafxError::StringError(
+                "Found more than one MaterialPass in MaterialAsset.".to_string(),
+            ))
+        }
+    }
+
+    pub fn get_material_pass_by_index(
+        &self,
+        index: usize,
+    ) -> Option<ResourceArc<MaterialPassResource>> {
+        self.inner
+            .passes
+            .get(index)
+            .map(|x| x.material_pass_resource.clone())
```

Comment:
Could probably do .get(index)?.clone()

Reply:
same for the two similar cases below

Reply:
This did not work, but I addressed everything else.